cmds/rmmod/rmmod.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2012-2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Remove a module from the Linux kernel
+//
+// Synopsis:
+//	rmmod name
+//
+// Description:
+//	rmmod is a clone of rmmod(8)
+//
+// Author:
+//     Roland Kammerer <dev.rck@gmail.com>
+package main
+
+import (
+	"log"
+	"os"
+	"syscall"
+	"unsafe"
+)
+
+func main() {
+	if len(os.Args) < 2 {
+		log.Fatalf("rmmod: ERROR: missing module name.\n")
+	}
+
+	modname := os.Args[1]
+	flags := syscall.O_NONBLOCK
+	ret, _, err := syscall.Syscall(syscall.SYS_DELETE_MODULE, uintptr(unsafe.Pointer(&[]byte(modname)[0])), uintptr(len(modname)), uintptr(unsafe.Pointer(&flags)))
+	if ret != 0 {
```

Comment:
Check err != Nil instead

Reply:
Hm, if you like. Again, taken from the existing "insmod", and I saw that kind of ret-checking for syscall in the wild. Maybe this would then be an extra patch.